torchvision/transforms/autoaugment.py

```diff
@@ -0,0 +1,232 @@
+import math
+import torch
+
+from enum import Enum
+from torch import Tensor
+from torch.jit.annotations import List, Tuple
+from typing import Optional
+
+from . import functional as F
+
+
+class AutoAugmentPolicy(Enum):
+    """AutoAugment policies learned on different datasets.
+    """
+    IMAGENET = "imagenet"
+    CIFAR10 = "cifar10"
+    SVHN = "svhn"
+
+
+class AutoAugment(torch.nn.Module):
+    r"""AutoAugment data augmentation method based on
+    `"AutoAugment: Learning Augmentation Strategies from Data" <https://arxiv.org/pdf/1805.09501.pdf>`_.
+    The image can be a PIL Image or a Tensor, in which case it is expected
+    to have [..., H, W] shape, where ... means an arbitrary number of leading dimensions.
+
+    Args:
+        policy (AutoAugmentPolicy): Desired policy enum defined by
+            :class:`torchvision.transforms.autoaugment.AutoAugmentPolicy`. Default is ``AutoAugmentPolicy.IMAGENET``.
+        fill (sequence or int or float, optional): Pixel fill value for the area outside the transformed
+            image. If int or float, the value is used for all bands respectively.
+            This option is supported for PIL image and Tensor inputs.
+            If input is PIL Image, the options is only available for ``Pillow>=5.0.0``.
+
+    Example:
+        >>> t = transforms.AutoAugment()
+        >>> transformed = t(image)
+
+        >>> transform=transforms.Compose([
+        >>>     transforms.Resize(256),
+        >>>     transforms.AutoAugment(),
+        >>>     transforms.ToTensor()])
+    """
+
+    def __init__(self, policy: AutoAugmentPolicy = AutoAugmentPolicy.IMAGENET, fill: Optional[List[float]] = None):
+        super().__init__()
+        self.policy = policy
+        self.fill = fill
+        if policy == AutoAugmentPolicy.IMAGENET:
```

Comment:
Maybe it would be nice to define predefined policies outside the class for better code reading.
Also, probably, it would be helpful to be able to customly define them too, like
```python
transforms.AutoAugment(policy=AutoAugmentPolicy.IMAGENET)
# or 
my_policy = [
(("Posterize", 0.4, 8), ("Rotate", 0.6, 9), ("Solarize", 0.6, 5)),
(("Equalize", 0.4, 8), ("Color", 0.6, 9), ("Solarize", 0.6, 5)),
]
transforms.AutoAugment(policy=my_policy)
```

Reply:
The original plan was to have it on a separate variable and pass it. Unfortunately JIT did not like that.

Also note that the paper implements those 3 specific policies only, so these are pretty fixed.

Reply:
Yes, that's correct that global vars wont be accepted by JIT. Maybe, policy definition can be put into enum AutoAugmentPolicy.IMAGENET somehow ?

> Also note that the paper implements those 3 specific policies only, so these are pretty fixed.

That's correct, we have 3 fixed policies and we can imagine that folks could learn somehow other policies for other datasets and apply them with torchvision... Policy values also probably guaranteed for PIL only, not tensors...

Reply:
Done! I moved it on a separate function similar to:
https://github.com/pytorch/vision/blob/aa753263b8e7b3180225e3ad1e6e5434a5f42882/torchvision/transforms/functional.py#L37-L46